src/codegen/llvm/codegen_llvm_helper_visitor.cpp

```diff
@@ -98,26 +103,18 @@ std::shared_ptr<ast::Expression> create_expression(const std::string& code) {
  *
  * We perform following transformations so that code generation backends
  * will have minimum logic:
- *  - Add return type
  *  - Add type for the function arguments
  *  - Define variables and return variable
- *  - Add return type (int for PROCEDURE and double for FUNCTION)
+ *  - Add return type (void for PROCEDURE and double for FUNCTION)
  */
 void CodegenLLVMHelperVisitor::create_function_for_node(ast::Block& node) {
     /// name of the function from the node
     std::string function_name = node.get_node_name();
     auto name = new ast::Name(new ast::String(function_name));
 
-    /// return variable name has "ret_" prefix
+    /// return variable name has "ret_" prefix and has type double
     auto return_var = new ast::Name(new ast::String("ret_" + function_name));
-
-    /// return type based on node type
-    ast::CodegenVarType* ret_var_type = nullptr;
-    if (node.get_node_type() == ast::AstNodeType::FUNCTION_BLOCK) {
-        ret_var_type = new ast::CodegenVarType(FLOAT_TYPE);
-    } else {
-        ret_var_type = new ast::CodegenVarType(INTEGER_TYPE);
-    }
+    auto ret_var_type = new ast::CodegenVarType(FLOAT_TYPE);
```

Comment:
Curious why you changed the type from int to float for PROCEDURE. It's true that they currently return just value 0. But their actual goal is to return error code which we are not doing now.

Reply:
There was no any specific reason - I think it was done in LLVM visitor like that before - I just found it more natural since then we do not need to specifically initialise the return variable. But with error codes it all makes sense!

Reply:
I will adjust the PR.